src/cryptoadvance/specter/node.py

```diff
@@ -534,6 +535,35 @@ def check_blockheight(self):
     def is_liquid(self):
         return is_liquid(self.chain)
 
+    def delete_wallet_file(self, wallet) -> bool:
+        """Deleting the wallet file located on the node. This only works if the node is on the same machine as Specter.
+        Returns True if the wallet file could be deleted, otherwise returns False."""
+        datadir = ""
+        if self.datadir == "":
+            # In case someone did not toggle the auto-detect but still used the default location.
+            # When you set up a new node and deactivate the auto-detect, the datadir is set to an empty string.
+            logger.debug(
+                f"The node datadir before get_default_datadir is: {self.datadir}"
+            )
+            datadir = get_default_datadir(self.node_type)
+            logger.debug(f"The node datadir after get_default_datadir is: {datadir}")
+        else:
+            datadir = self.datadir
+        wallet_file_removed = False
+        path = ""
+        wallet_rpc_path = os.path.join(wallet.manager.rpc_path, wallet.alias)
```

Comment:
```suggestion
        wallet_rpc_path = os.path.join(wallet.manager.rpc_path, wallet.alias)
        try:
            wallet.rpc.unloadwallet(wallet_rpc_path)
            raise SpecterError("Trying to delete a Walletfile but wallet was not unloaded corectly!")
        except RpcError:
            pass
```
Make sure that the wallet is REALLY unloaded. If you delete the file while the wallet in still loaded, you're breaking Bitcoin Core which in turn results in  `BrokenCoreConnectionException`.

Reply:
Shouldn't the the SpecterError be raised in the RpcError?

Reply:
Apart from that, good point!

Reply:
Thanks for the input, resolved in https://github.com/cryptoadvance/specter-desktop/pull/1950/commits/8e3adaf369c5b8da9f6bb238d1d61c93ce28fc70